src/engine/controls/loopingcontrol.cpp

```diff
@@ -965,7 +965,8 @@ void LoopingControl::slotLoopStartPos(double positionSamples) {
     loopInfo.startPosition = position;
     m_pCOLoopStartPosition->set(position.toEngineSamplePosMaybeInvalid());
 
-    if (loopInfo.endPosition.isValid() && loopInfo.endPosition <= loopInfo.startPosition) {
+    if (loopInfo.startPosition.isValid() && loopInfo.endPosition.isValid() &&
```

Comment:
We already checked for `position.isValid()` above. I don't see a reason why we cannot combine these code paths like `if (!loopInfo.startPosition.isValid() || !loopInfo.endPosition().isValid() || loopInfo.endPosition <= loopInfo.startPosition) ...`?

Reply:
https://github.com/mixxxdj/mixxx/pull/4266/commits/d8f91a9cebe4c365f4e36d79d0e205d3e22580fb